src/trustRegion.jl

```diff
@@ -140,6 +140,11 @@ mutable struct TrustRegionCache{iip, fType, algType, uType, resType, pType,
     prob::probType
     trust_r::trustType
     max_trust_r::trustType
+    step_threshold::trustType
```

Comment:
```suggestion
    step_threshold::suType
```

Reply:
I don't think this will work, `step_threshold` is a scalar and `suType` seems to be a vector.

I suppose https://github.com/SciML/NonlinearSolve.jl/pull/135 was for a good reason, but when I try the program with
```julia
step_size::uType
    u_tmp::uType
```
and not the new types, the first execution is about 4 times faster...
Will look more into the timing now, but is it necessary to have the new types?

Reply:
(and `step_threshold::trustType`)

Reply:
Oh okay if you're back cool. I was just poking at it through quick commits and checking CI results seeing if I can get this solved while hacking on something else, But if you have a second to push this over the line, it probably just needs a different type parameter. I cannot say I was looking too closely.

Reply:
Yeah, I've got some time now. I'll probably have to make a PR to daviehh (original PR creator) but hopefully, he accepts it soon enough.